specs/dwritecore/DWriteCore.md

```diff
@@ -170,8 +412,469 @@ DWRITE_BITMAP_DATA_BGRA32 TextRenderer::GetBitmapData(_In_ IDWriteBitmapRenderTa
 }
 ```
 
+## Using Font Selection APIs
+
+This section shows a complete console application that demonstrates the `IDWriteFontSet4::GetMatchingFonts`
+and `IDWriteFontSet4::GetDerivedFontAxisValues` methods. First let's include some headers:
+
+```c++
+#include <dwrite_core.h>
+#include <wil/com.h>
+#include <iostream>
+#include <string>
+#include <vector>
+```
+
+The `IDWriteFontSet4::GetMatchingFonts` method returns a list of fonts in priority order that match the
+specified family name and axis values. The following `MatchAxisValues` function outputs the parameters 
+to `GetMatchingFonts` and the list of matching fonts in the returned font set.
+
+```c++
+// Forward declarations of overloaded output operators used by MatchAxisValues.
+std::wostream& operator<<(std::wostream& out, DWRITE_FONT_AXIS_VALUE axisValue);
+std::wostream& operator<<(std::wostream& out, IDWriteFontFile& fileReference);
+std::wostream& operator<<(std::wostream& out, IDWriteFontFaceReference1& faceReference);
+
+//
+// MatchAxisValues calls IDWriteFontSet4::GetMatchingFonts with the
+// specific parameters and outputs the matching fonts.
+//
+void MatchAxisValues(
+    IDWriteFontSet4* fontSet,
+    _In_z_ WCHAR const* familyName,
+    std::vector<DWRITE_FONT_AXIS_VALUE> const& axisValues,
+    bool allowSimulations
+    )
+{
+    // Write the input parameters.
+    std::wcout << L"GetMatchingFonts(\"" << familyName << L"\", {";
+    for (DWRITE_FONT_AXIS_VALUE axisValue : axisValues)
+    {
+        std::wcout << L' ' << axisValue;
+    }
+    std::wcout
+        << L" }, "
+        << (allowSimulations ? L"true" : L"false")
+        << L"):\n";
+
+    // Get the matching fonts for the specified family name and axis values.
+    wil::com_ptr<IDWriteFontSet4> matchingFonts;
+    THROW_IF_FAILED(fontSet->GetMatchingFonts(
+        familyName,
+        axisValues.data(),
+        static_cast<UINT32>(axisValues.size()),
+        allowSimulations,
```

Comment:
As discussed in emails, it might make sense to have allowSimulation be an enum or bitmask to control what simulation type should be disallowed. (compare CSS font-synthesis property), i.e. only allowing italic synthetisation, only allowing bold synthetisation, (small-caps?).

Reply:
Sounds reasonable. (DWrite doesn't support small cap simulation.)